pkg/util/cloudprovider.go

```diff
@@ -0,0 +1,41 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2019 Datadog, Inc.
+
+package util
+
+import (
+	"github.com/DataDog/datadog-agent/pkg/metadata/inventories"
+	"github.com/DataDog/datadog-agent/pkg/util/alibaba"
+	"github.com/DataDog/datadog-agent/pkg/util/azure"
+	"github.com/DataDog/datadog-agent/pkg/util/ec2"
+	"github.com/DataDog/datadog-agent/pkg/util/ecs"
+	"github.com/DataDog/datadog-agent/pkg/util/gce"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// DetectCloudProvider detects the cloud provider where the agent is running in order:
+// * AWS ECS/Fargate
+// * AWS EC2
+// * GCE
+// * Azure
+// * Alibaba
+func DetectCloudProvider() {
+	detectors := map[string]func() bool{
+		ecs.CloudProviderName:     ecs.IsRunningOn,
+		ec2.CloudProviderName:     ec2.IsRunningOn,
+		gce.CloudProviderName:     gce.IsRunningOn,
+		azure.CloudProviderName:   azure.IsRunningOn,
+		alibaba.CloudProviderName: alibaba.IsRunningOn,
+	}
+
+	for cloudProviderName, cloudDetector := range detectors {
+		if cloudDetector() {
+			inventories.SetAgentMetadata(inventories.CloudProviderMetatadaName, cloudProviderName)
```

Comment:
question for later: do we have some logic somewhere to only send the inventories payload once all the initial metadata has been set? So that we don't send the payload on the first call to `inventories.SetAgentMetadata`, and instead wait a bit for the first metadata collection to be "complete".

Reply:
We don't have the logic to know when the inventory payload is "complete". But the first payload will be sent 5 minutes after the agent starts, at this time inventory should have received all its info